webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/facts/nodes/NodeFact.scala

```diff
@@ -0,0 +1,1028 @@
+      accessor(a).size == accessor(b).size &&
+      accessor(a).sortBy(orderOn) == accessor(b).sortBy(orderOn)
+    }
+
+    def eq[A](accessor: NodeFact => A): Boolean = {
+      accessor(a) == accessor(b)
+    }
+
+    eq(_.id) &&
+    eq(_.description) &&
+    eq(_.fqdn) &&
+    eq(_.os) &&
+    eq(_.machine) &&
+    eq(_.rudderSettings) &&
+    eq(_.rudderAgent) &&
+    compare(_.properties)(_.name) &&
+    compare(_.ipAddresses)(_.inet) &&
+    eq(_.timezone) &&
+    eq(_.ram) &&
+    eq(_.swap) &&
+    eq(_.archDescription) &&
+    compare(_.accounts)(identity) &&
+    compare(_.bios)(_.name) &&
+    compare(_.controllers)(_.name) &&
+    compare(_.environmentVariables)(_._1) &&
+    compare(_.fileSystems)(_.mountPoint) &&
+    compare(_.inputs)(_.tpe) &&
+    compare(_.localGroups)(_.id) &&
+    compare(_.localUsers)(_.id) &&
+    compare(_.logicalVolumes)(_.lvUUID) &&
+    compare(_.memories)(_.slotNumber) &&
+    compare(_.networks)(_.name) &&
+    compare(_.physicalVolumes)(_.pvUUID) &&
+    compare(_.ports)(_.name) &&
+    compare(_.processes)(_.pid) &&
+    compare(_.processors)(_.name) &&
+    compare(_.slots)(_.name) &&
+    compare(_.software)(_.name) &&
+    compare(_.softwareUpdate)(_.name) &&
+    compare(_.sounds)(_.name) &&
+    compare(_.storages)(_.name) &&
+    compare(_.videos)(_.name) &&
+    compare(_.vms)(_.uuid.value)
+  }
+
+  def sortAttributes(node: NodeFact): NodeFact = {
+    node
+      .modify(_.properties)
+      .using(_.sortBy(_.name))
+      .modify(_.ipAddresses)
+      .using(_.sortBy(_.inet))
+      .modify(_.accounts)
+      .using(_.sorted)
+      .modify(_.bios)
+      .using(_.sortBy(_.name))
+      .modify(_.controllers)
+      .using(_.sortBy(_.name))
+      .modify(_.environmentVariables)
+      .using(_.sortBy(_._1))
+      .modify(_.inputs)
+      .using(_.sortBy(_.tpe))
+      .modify(_.fileSystems)
+      .using(_.sortBy(_.name))
+      .modify(_.localGroups)
+      .using(_.sortBy(_.id))
+      .modify(_.localUsers)
+      .using(_.sortBy(_.id))
+      .modify(_.logicalVolumes)
+      .using(_.sortBy(_.lvName))
+      .modify(_.memories)
+      .using(_.sortBy(_.slotNumber))
+      .modify(_.networks)
+      .using(_.sortBy(_.name))
+      .modify(_.ports)
+      .using(_.sortBy(_.name))
+      .modify(_.processes)
+      .using(_.sortBy(_.pid))
+      .modify(_.processors)
+      .using(_.sortBy(_.name))
+      .modify(_.slots)
+      .using(_.sortBy(_.name))
+      .modify(_.software)
+      .using(_.sortBy(_.name))
+      .modify(_.softwareUpdate)
+      .using(_.sortBy(_.name))
+      .modify(_.sounds)
+      .using(_.sortBy(_.name))
+      .modify(_.storages)
+      .using(_.sortBy(_.name))
+      .modify(_.videos)
+      .using(_.sortBy(_.name))
+      .modify(_.vms)
+      .using(_.sortBy(_.name))
+  }
+
+  def toMachineId(nodeId: NodeId) = MachineUuid("machine-" + nodeId.value)
+
+  implicit class IterableToChunk[A](it: Iterable[A]) {
```

Comment:
Should not this be in a more generic file in utils ? where we have all things related to implicit conversion of basic datastrcutures ?

Reply:
Perhaps, I wasn't sure why it was not in the base encoder (same for either). I wonder if it doesn't create implicit resolution divergence in the general case. I can try to move it there

Reply:
I propose to look at that in a subsequent PR, because perhaps there is other such case already existing, we should merge them all in `utils` (I'm thinking to datetime, etc)